irods/client_configuration/__init__.py

```diff
@@ -145,7 +145,7 @@ class _ConfigLoadError:
 class NoConfigError(Exception, _ConfigLoadError): pass
 class BadConfigError(Exception, _ConfigLoadError): pass
 
-def load(root = None, file = '', failure_modes = (), logging_level = logging.WARNING):
+def load(root = None, file = '', failure_modes = (), do_load = True, logging_level = logging.WARNING):
```

Comment:
I think we could rename this option to make it a little clearer. `do_load` as an option to a function named `load` is confusing, at least to me. Perhaps `detect_only=False` or `modify_config=True` or something like that could help.

Better yet, we could separate out the first part of this function into some `open_configuration` function or something like that and have `load` call that and then have a third function to `detect_configuration`. That might be a bit more heavy lifting than it's worth.

Thoughts?

Reply:
> I think we could rename this option to make it a little clearer. `do_load` as an option to a function named `load` is confusing, at least to me. Perhaps `detect_only=False` or `modify_config=True` or something like that could help.

Definitely.

> Better yet, we could separate out the first part of this function into some `open_configuration` function or something like that and have `load` call that and then have a third function to `detect_configuration`. That might be a bit more heavy lifting than it's worth.
> 
> Thoughts?

Sounds reasonable. Especially if it helps with readability and maintenance.

Reply:
Done